### DISCUSSION TITLE
Releated posts in other themes

### DISCUSSION THREAD

**Discussion by KuraMiYanko** [bootscore/bootscore#547] [Q&A]
Since i am not using your theme, is is possible to change the hook for related posts, to a hook of my choice ?

**Comment by crftwrk** [comment]
Converted this to a discussion. Sure, it's possible.

1. Override related post template in your child as described here https://bootscore.me/documentation/bs-swiper/#Overriding_templates_via_theme

2. Open childs `related-posts.php` and go to line 14:

```php
function bootscore_related_posts() {
```

3. Replace this line with your function name:

```php
function my_custom_related_posts() {
```

4. Add new hook to your `single.php`:

```php
<?php my_custom_related_posts(); ?>
```

That's it. But note that all other templates in this plugin have dependencies to bootScore's functions. If you want to use them in a different theme, those dependencies must be removed:

```php
<?php bootscore_category_badge(); ?>
```

```php
<?php if ('post' === get_post_type()) : ?>
  <p class="meta small mb-2 text-muted">
    <?php
      bootscore_date();
      bootscore_author();
      bootscore_comments();
      bootscore_edit();
    ?>
  </p>
<?php endif; ?>
```

```php
<?php bootscore_tags(); ?>
```

Solved?

**Comment by KuraMiYanko** [comment]
Hi, 
thank you for the replay. I am not sure what i'm doing wrong. I have changed the hook to an already existing one 
`function woocommerce_after_template_part() {` but the related posts is not working. I have even tried to change the main template file.

I have another question, i see that the swiper does take the woo template, but it does not inherit the css, so iam guessing that there are difference in the classes. How can i change the classes to the defaul woocommerce ones ?

**Reply by crftwrk** [reply]
Related posts works only for posts in single.php. `function woocommerce_after_template_part() {` seems that you want to add this somewhere else, right? This won't work. If you explain what you exactly want to do, I can answer if that is possible or not.

`sc-swiper-card-product.php` https://github.com/bootscore/bs-swiper/blob/main/templates/sc-swiper-card-product.php uses a modified loop of WooCommerce loop https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/templates/content-product.php.

Override `sc-swiper-card-product.php` in your child as described above and change HTML to the default WooCommerce.

**Comment by KuraMiYanko** [comment]
Hi, i have realised that i was using the wrong hook, so i have changed it to this - `function woocommerce_after_single_product_summary() {`
but still nothing is happening. I am trying to use it on the single product page. I have woo default releated products at the moment, not shure if that can create a conflict.

For the `sc-swiper-card-product.php` this is how it looks, after i changed the HTML
` <?php

/*
 *
 * Product slider template.
 *
 * This template can be overriden by copying this file to your-theme/bs-swiper-main/sc-swiper-card-product.php
 *
 * @author   bootScore
 * @package  bS Swiper
 * @version  5.3.0
 *
 * Product Slider Shortcode
 * [bs-swiper-card-product order="DESC" orderby="date" posts="12" category="theme, child-themes, free, plugins"]
 *
*/


// Product Slider Shortcode
add_shortcode('bs-swiper-card-product', 'bootscore_product_slider');
function bootscore_product_slider($atts) {

  ob_start();
  extract(shortcode_atts(array(
    'type' => 'product',
    'order' => 'date',
    'orderby' => 'date',
    'posts' => -1,
    'category' => '',
  ), $atts));

  $options = array(
    'order' => $order,
    'orderby' => $orderby,
    'posts_per_page' => $posts,
    'product_cat'    => $category,
  );

  $query = new WP_Query($options);
  if ($query->have_posts()) { ?>


    

    <div class="px-5 position-relative product-slider woocommerce">

      <div class="cards swiper-container swiper position-static">

        <div class="swiper-wrapper">

          <?php while ($query->have_posts()) : $query->the_post(); ?>

            <?php


defined( 'ABSPATH' ) || exit;

global $product;

// Ensure visibility.
if ( empty( $product ) || ! $product->is_visible() ) {
  return;
}
?>
<li <?php wc_product_class( '', $product ); ?>>
  <?php
  /**
   * Hook: woocommerce_before_shop_loop_item.
   *
   * @hooked woocommerce_template_loop_product_link_open - 10
   */
  do_action( 'woocommerce_before_shop_loop_item' );

  /**
   * Hook: woocommerce_before_shop_loop_item_title.
   *
   * @hooked woocommerce_show_product_loop_sale_flash - 10
   * @hooked woocommerce_template_loop_product_thumbnail - 10
   */
  do_action( 'woocommerce_before_shop_loop_item_title' );

  /**
   * Hook: woocommerce_shop_loop_item_title.
   *
   * @hooked woocommerce_template_loop_product_title - 10
   */
  do_action( 'woocommerce_shop_loop_item_title' );

  /**
   * Hook: woocommerce_after_shop_loop_item_title.
   *
   * @hooked woocommerce_template_loop_rating - 5
   * @hooked woocommerce_template_loop_price - 10
   */
  do_action( 'woocommerce_after_shop_loop_item_title' );

  /**
   * Hook: woocommerce_after_shop_loop_item.
   *
   * @hooked woocommerce_template_loop_product_link_close - 5
   * @hooked woocommerce_template_loop_add_to_cart - 10
   */
  do_action( 'woocommerce_after_shop_loop_item' );
  ?>
</li>

          <?php endwhile;
          wp_reset_postdata(); ?>

        </div> 

        
        <div class="swiper-pagination"></div>
        
        <div class="swiper-button-next end-0"></div>
        <div class="swiper-button-prev start-0"></div>

      </div>


    </div>

    

<?php $myvariable = ob_get_clean();
    return $myvariable;
  }
}

// Product Slider Shortcode End `
but its not working, so i am shure thats not it

**Reply by crftwrk** [reply]
As I mentioned above, the related posts works only for posts in single.php, not for products ;-)

**Comment by KuraMiYanko** [comment]
I understand now, thank you for your time :)

**Reply by crftwrk** [reply]
You‘re welcome 👍
